protocols/raft/src/main/java/io/atomix/protocols/raft/RaftServer.java

```diff
@@ -514,6 +514,11 @@ default boolean isLeader() {
    */
   boolean isRunning();
 
+  /**
+   * Steps down from the current leadership, which means tries to transition directly to follower.
+   */
+  void stepDown();
```

Comment:
It would be more consistent with the API to return a future here, e.g. `CompletableFuture<RaftServer> promote();` or `CompletableFuture<Void> shutdown();`

Is there any reason we wouldn't want to do that?